<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction to Single Cell Technology | Fundamentals of scRNASeq Analysis</title>
  <meta name="description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction to Single Cell Technology | Fundamentals of scRNASeq Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="github-repo" content="linxy29/Fundamental-scRNA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction to Single Cell Technology | Fundamentals of scRNASeq Analysis" />
  
  <meta name="twitter:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  

<meta name="author" content="Xinyi Lin, Gordon Qian, Joshua Ho contact: jwkho@hku.hk" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="raw2matrix.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Fundamental scRNASeq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prerequisites</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-data"><i class="fa fa-check"></i>Download data</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Introduction to Single Cell Technology</a><ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#single-cell-rna-sequencing-scrna-seq"><i class="fa fa-check"></i><b>1.1</b> Single cell RNA sequencing (scRNA-seq)</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#cell-isolation"><i class="fa fa-check"></i><b>1.2</b> Cell isolation</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#barcodes-and-unique-molecular-identifiers-umi"><i class="fa fa-check"></i><b>1.3</b> Barcodes and Unique molecular identifiers (UMI)</a></li>
<li class="chapter" data-level="1.4" data-path="background.html"><a href="background.html#summary-of-widely-used-scrna-seq-technologies"><i class="fa fa-check"></i><b>1.4</b> Summary of widely used scRNA-seq technologies</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="raw2matrix.html"><a href="raw2matrix.html"><i class="fa fa-check"></i><b>2</b> Pre-processing and QC</a><ul>
<li class="chapter" data-level="2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#fastq-file"><i class="fa fa-check"></i><b>2.1</b> FASTQ file</a></li>
<li class="chapter" data-level="2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#cell-ranger"><i class="fa fa-check"></i><b>2.2</b> Cell Ranger</a><ul>
<li class="chapter" data-level="2.2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#general-workflow"><i class="fa fa-check"></i><b>2.2.1</b> General workflow</a></li>
<li class="chapter" data-level="2.2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#output-file"><i class="fa fa-check"></i><b>2.2.2</b> Output file</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="raw2matrix.html"><a href="raw2matrix.html#starsolo"><i class="fa fa-check"></i><b>2.3</b> STARsolo</a></li>
<li class="chapter" data-level="2.4" data-path="raw2matrix.html"><a href="raw2matrix.html#doublets"><i class="fa fa-check"></i><b>2.4</b> Doublets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downstream.html"><a href="downstream.html"><i class="fa fa-check"></i><b>3</b> Analysis Using Seurat</a><ul>
<li class="chapter" data-level="3.1" data-path="downstream.html"><a href="downstream.html#seurat-object"><i class="fa fa-check"></i><b>3.1</b> Seurat object</a></li>
<li class="chapter" data-level="3.2" data-path="downstream.html"><a href="downstream.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="downstream.html"><a href="downstream.html#standard-pre-processing-workflow"><i class="fa fa-check"></i><b>3.3</b> Standard pre-processing workflow</a><ul>
<li class="chapter" data-level="3.3.1" data-path="downstream.html"><a href="downstream.html#qc-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>3.3.1</b> QC and selecting cells for further analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="downstream.html"><a href="downstream.html#normalizing-the-data"><i class="fa fa-check"></i><b>3.3.2</b> Normalizing the data</a></li>
<li class="chapter" data-level="3.3.3" data-path="downstream.html"><a href="downstream.html#identification-of-highly-variable-features-feature-selection"><i class="fa fa-check"></i><b>3.3.3</b> Identification of highly variable features (feature selection)</a></li>
<li class="chapter" data-level="3.3.4" data-path="downstream.html"><a href="downstream.html#scaling-the-data"><i class="fa fa-check"></i><b>3.3.4</b> Scaling the data</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="downstream.html"><a href="downstream.html#perform-linear-dimensional-reduction"><i class="fa fa-check"></i><b>3.4</b> Perform linear dimensional reduction</a></li>
<li class="chapter" data-level="3.5" data-path="downstream.html"><a href="downstream.html#cluster-the-cells"><i class="fa fa-check"></i><b>3.5</b> Cluster the cells</a></li>
<li class="chapter" data-level="3.6" data-path="downstream.html"><a href="downstream.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>3.6</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fundamentals of scRNASeq Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction to Single Cell Technology</h1>
<div id="single-cell-rna-sequencing-scrna-seq" class="section level2">
<h2><span class="header-section-number">1.1</span> Single cell RNA sequencing (scRNA-seq)</h2>
<p>Single cell RNA sequencing (scRNA-seq) encompasses a range of technology to generate genome-wide expression of many individual single cells.</p>
<p>Starting from tissue samples we get, we will get the analysis results after following processes.</p>
<p><br/></p>
<div class="figure" style="text-align: center"><span id="fig:workflow"></span>
<img src="figs/workflow.jpg" alt="Single Cell Genome Sequencing Workflow (taken from Wiki). MDA: multiple displacement amplification" width="80%" />
<p class="caption">
Figure 1.1: Single Cell Genome Sequencing Workflow (taken from Wiki). MDA: multiple displacement amplification
</p>
</div>
<p><br/></p>
</div>
<div id="cell-isolation" class="section level2">
<h2><span class="header-section-number">1.2</span> Cell isolation</h2>
<p>Droplet-based:</p>
<ul>
<li><p>tissue sample must be dissociated into suspension</p></li>
<li><p>cells will be encapsulated into a water-in-oildroplet individually</p></li>
<li><p>high-throughput and low cost</p></li>
<li><p>related technologies: Drop-seq <span class="citation">(Macosko et al. <a href="#ref-macosko2015highly" role="doc-biblioref">2015</a>)</span>, inDrop <span class="citation">(Klein et al. <a href="#ref-klein2015droplet" role="doc-biblioref">2015</a>)</span>, Chromium 10X <span class="citation">(Zheng et al. <a href="#ref-zheng2017massively" role="doc-biblioref">2017</a>)</span></p></li>
</ul>
<p><br/></p>
<p>Non droplet-based:</p>
<ul>
<li><p>Smart-seq2 <span class="citation">(Ramsköld et al. <a href="#ref-ramskold2012full" role="doc-biblioref">2012</a>)</span>: manual cell picking with micro capillary pipettes</p></li>
<li><p>CEL-seq <span class="citation">(Hashimshony et al. <a href="#ref-hashimshony2012cel" role="doc-biblioref">2012</a>)</span>: individual cells are added to tubes; the first one introducing barcodes and pooling of RNA</p></li>
<li><p>MARS-seq <span class="citation">(Jaitin et al. <a href="#ref-jaitin2014massively" role="doc-biblioref">2014</a>)</span> is the first one using FACS to isolate single cells into individual wells, and the optimized version MARS-seq2 <span class="citation">(Keren-Shaul et al. <a href="#ref-keren2019mars" role="doc-biblioref">2019</a>)</span> came out with lower cost, improved reproducibility, reduced well-to-well contamination</p></li>
</ul>
<p><br/></p>
<p>The following figure shows some common single cell isolaion techniques <span class="citation">(Hwang, Lee, and Bang <a href="#ref-hwang2018single" role="doc-biblioref">2018</a>)</span>.</p>
<div class="figure" style="text-align: center"><span id="fig:seq-techs"></span>
<img src="figs/seq_techs.jpg" alt="Single Cell Isolation (modified from Hwang, Lee, and Bang 2018)" width="80%" />
<p class="caption">
Figure 1.2: Single Cell Isolation (modified from Hwang, Lee, and Bang 2018)
</p>
</div>
<p>For more information about different cell isolation techniques: <span class="citation">Hu et al. (<a href="#ref-hu2016single" role="doc-biblioref">2016</a>)</span></p>
</div>
<div id="barcodes-and-unique-molecular-identifiers-umi" class="section level2">
<h2><span class="header-section-number">1.3</span> Barcodes and Unique molecular identifiers (UMI)<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></h2>
<p>This part is modified based on the lecture note, <a href="http://data-science-sequencing.github.io/Win2018/lectures/lecture16/">Single Cell RNA-Seq - Introduction</a> created by David <span class="citation">Tse (<a href="#ref-tse_2018" role="doc-biblioref">2018</a>)</span>.</p>
<p>Paired-end sequencing outputs two fastq files corresponding to the 5’ and 3’ direction of sequencing. With this sequencing technology, the first read of the pair always coincides with the cell (barcode + UMI) part of the primer.</p>
<div class="figure" style="text-align: center"><span id="fig:biasedreads"></span>
<img src="figs/biasedreads.png" alt="Biased paired-end reads (David Tse et.al)" width="80%" />
<p class="caption">
Figure 1.3: Biased paired-end reads (David Tse et.al)
</p>
</div>
<p>Based on the obtained reads consisting of cell barcode, UMI and cDNA, we can estimate the transcript abundances. This allows the mapping algorithm to distinguish which sequences are barcodes and which are transcript sequences. Thus, it is important to recognize the library preparation chemistry used for sequencing in order to determine cell barcode and UMI barcode sequence length and location.</p>
<p>To get the UMIS’ couns, we can first group reads by cell barcode before aligning cDNA reads and counting unique molecules per cell per gene using the UMIs.</p>
<div class="figure" style="text-align: center"><span id="fig:biasedreads2"></span>
<img src="figs/biasedreads2.png" alt="Grouping barcodes to assign reads to cells (modified from David Tse et.al)" width="70%" />
<p class="caption">
Figure 1.4: Grouping barcodes to assign reads to cells (modified from David Tse et.al)
</p>
</div>
<p>Analysis of the cell barcodes and UMIs is included in the alignment process, and we will introduce more in Chapter <a href="raw2matrix.html#raw2matrix">2</a>.</p>
<pre><code>**Learning Assessment**
1. What is the difference between a cell barcode and UMI barcode? and what are their significance?
2. What are the lengths of the cell barcode and UMI barcode used in our dataset?
3. Are these barcodes located on the 5&#39; or 3&#39; read file?</code></pre>
</div>
<div id="summary-of-widely-used-scrna-seq-technologies" class="section level2">
<h2><span class="header-section-number">1.4</span> Summary of widely used scRNA-seq technologies</h2>
<p>Following is a summary table from <span class="citation">(Chen, Ning, and Shi <a href="#ref-chen2019single" role="doc-biblioref">2019</a>)</span>. It shows different features of widely used scRNA-seq technologies.</p>
<table>
<colgroup>
<col width="15%" />
<col width="22%" />
<col width="16%" />
<col width="12%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Methods</th>
<th>Transcript
coverage</th>
<th>UMI
possibility</th>
<th>Strand
specific</th>
<th>References</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Tang method</td>
<td>Nearly full-length</td>
<td>No</td>
<td>No</td>
<td><span class="citation">Tang et al. (<a href="#ref-tang2009mrna" role="doc-biblioref">2009</a>)</span></td>
</tr>
<tr class="even">
<td>Quartz-Seq</td>
<td>Full-length</td>
<td>No</td>
<td>No</td>
<td><span class="citation">Sasagawa et al. (<a href="#ref-sasagawa2013quartz" role="doc-biblioref">2013</a>)</span></td>
</tr>
<tr class="odd">
<td>SUPeR-seq</td>
<td>Full-length</td>
<td>No</td>
<td>No</td>
<td><span class="citation">X. Fan et al. (<a href="#ref-fan2015single" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td>Smart-seq</td>
<td>Full-length</td>
<td>No</td>
<td>No</td>
<td><span class="citation">Ramsköld et al. (<a href="#ref-ramskold2012full" role="doc-biblioref">2012</a>)</span></td>
</tr>
<tr class="odd">
<td>Smart-seq2</td>
<td>Full-length</td>
<td>No</td>
<td>No</td>
<td><span class="citation">Picelli et al. (<a href="#ref-picelli2013smart" role="doc-biblioref">2013</a>)</span></td>
</tr>
<tr class="even">
<td>MATQ-seq</td>
<td>Full-length</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Sheng et al. (<a href="#ref-sheng2017effective" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="odd">
<td>STRT-seq
STRT/C1</td>
<td>5′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Islam et al. (<a href="#ref-islam2011characterization" role="doc-biblioref">2011</a>)</span></td>
</tr>
<tr class="even">
<td>CEL-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Hashimshony et al. (<a href="#ref-hashimshony2012cel" role="doc-biblioref">2012</a>)</span></td>
</tr>
<tr class="odd">
<td>CEL-seq2</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Hashimshony et al. (<a href="#ref-hashimshony2016cel" role="doc-biblioref">2016</a>)</span></td>
</tr>
<tr class="even">
<td>MARS-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Jaitin et al. (<a href="#ref-jaitin2014massively" role="doc-biblioref">2014</a>)</span></td>
</tr>
<tr class="odd">
<td>CytoSeq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">H. C. Fan, Fu, and Fodor (<a href="#ref-fan2015combinatorial" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td>Drop-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Macosko et al. (<a href="#ref-macosko2015highly" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="odd">
<td>InDrop</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Klein et al. (<a href="#ref-klein2015droplet" role="doc-biblioref">2015</a>)</span></td>
</tr>
<tr class="even">
<td>Chromium</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Zheng et al. (<a href="#ref-zheng2017massively" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="odd">
<td>SPLiT-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Rosenberg et al. (<a href="#ref-rosenberg2018single" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="even">
<td>sci-RNA-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Cao et al. (<a href="#ref-cao2017comprehensive" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="odd">
<td>Seq-Well</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Gierahn et al. (<a href="#ref-gierahn2017seq" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="even">
<td>DroNC-seq</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Habib et al. (<a href="#ref-habib2017massively" role="doc-biblioref">2017</a>)</span></td>
</tr>
<tr class="odd">
<td>Quartz-Seq2</td>
<td>3′-only</td>
<td>Yes</td>
<td>Yes</td>
<td><span class="citation">Sasagawa et al. (<a href="#ref-sasagawa2018quartz" role="doc-biblioref">2018</a>)</span></td>
</tr>
</tbody>
</table>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-cao2017comprehensive">
<p>Cao, Junyue, Jonathan S Packer, Vijay Ramani, Darren A Cusanovich, Chau Huynh, Riza Daza, Xiaojie Qiu, et al. 2017. “Comprehensive Single-Cell Transcriptional Profiling of a Multicellular Organism.” <em>Science</em> 357 (6352): 661–67.</p>
</div>
<div id="ref-chen2019single">
<p>Chen, Geng, Baitang Ning, and Tieliu Shi. 2019. “Single-Cell Rna-Seq Technologies and Related Computational Data Analysis.” <em>Frontiers in Genetics</em> 10: 317.</p>
</div>
<div id="ref-fan2015combinatorial">
<p>Fan, H Christina, Glenn K Fu, and Stephen PA Fodor. 2015. “Combinatorial Labeling of Single Cells for Gene Expression Cytometry.” <em>Science</em> 347 (6222).</p>
</div>
<div id="ref-fan2015single">
<p>Fan, Xiaoying, Xiannian Zhang, Xinglong Wu, Hongshan Guo, Yuqiong Hu, Fuchou Tang, and Yanyi Huang. 2015. “Single-Cell Rna-Seq Transcriptome Analysis of Linear and Circular Rnas in Mouse Preimplantation Embryos.” <em>Genome Biology</em> 16 (1): 1–17.</p>
</div>
<div id="ref-gierahn2017seq">
<p>Gierahn, Todd M, Marc H Wadsworth, Travis K Hughes, Bryan D Bryson, Andrew Butler, Rahul Satija, Sarah Fortune, J Christopher Love, and Alex K Shalek. 2017. “Seq-Well: Portable, Low-Cost Rna Sequencing of Single Cells at High Throughput.” <em>Nature Methods</em> 14 (4): 395–98.</p>
</div>
<div id="ref-habib2017massively">
<p>Habib, Naomi, Inbal Avraham-Davidi, Anindita Basu, Tyler Burks, Karthik Shekhar, Matan Hofree, Sourav R Choudhury, et al. 2017. “Massively Parallel Single-Nucleus Rna-Seq with Dronc-Seq.” <em>Nature Methods</em> 14 (10): 955–58.</p>
</div>
<div id="ref-hashimshony2016cel">
<p>Hashimshony, Tamar, Naftalie Senderovich, Gal Avital, Agnes Klochendler, Yaron De Leeuw, Leon Anavy, Dave Gennert, et al. 2016. “CEL-Seq2: Sensitive Highly-Multiplexed Single-Cell Rna-Seq.” <em>Genome Biology</em> 17 (1): 1–7.</p>
</div>
<div id="ref-hashimshony2012cel">
<p>Hashimshony, Tamar, Florian Wagner, Noa Sher, and Itai Yanai. 2012. “CEL-Seq: Single-Cell Rna-Seq by Multiplexed Linear Amplification.” <em>Cell Reports</em> 2 (3): 666–73.</p>
</div>
<div id="ref-hu2016single">
<p>Hu, Ping, Wenhua Zhang, Hongbo Xin, and Glenn Deng. 2016. “Single Cell Isolation and Analysis.” <em>Frontiers in Cell and Developmental Biology</em> 4: 116.</p>
</div>
<div id="ref-hwang2018single">
<p>Hwang, Byungjin, Ji Hyun Lee, and Duhee Bang. 2018. “Single-Cell Rna Sequencing Technologies and Bioinformatics Pipelines.” <em>Experimental &amp; Molecular Medicine</em> 50 (8): 1–14.</p>
</div>
<div id="ref-islam2011characterization">
<p>Islam, Saiful, Una Kjällquist, Annalena Moliner, Pawel Zajac, Jian-Bing Fan, Peter Lönnerberg, and Sten Linnarsson. 2011. “Characterization of the Single-Cell Transcriptional Landscape by Highly Multiplex Rna-Seq.” <em>Genome Research</em> 21 (7): 1160–7.</p>
</div>
<div id="ref-jaitin2014massively">
<p>Jaitin, Diego Adhemar, Ephraim Kenigsberg, Hadas Keren-Shaul, Naama Elefant, Franziska Paul, Irina Zaretsky, Alexander Mildner, et al. 2014. “Massively Parallel Single-Cell Rna-Seq for Marker-Free Decomposition of Tissues into Cell Types.” <em>Science</em> 343 (6172): 776–79.</p>
</div>
<div id="ref-keren2019mars">
<p>Keren-Shaul, Hadas, Ephraim Kenigsberg, Diego Adhemar Jaitin, Eyal David, Franziska Paul, Amos Tanay, and Ido Amit. 2019. “MARS-Seq2. 0: An Experimental and Analytical Pipeline for Indexed Sorting Combined with Single-Cell Rna Sequencing.” <em>Nature Protocols</em> 14 (6): 1841–62.</p>
</div>
<div id="ref-klein2015droplet">
<p>Klein, Allon M, Linas Mazutis, Ilke Akartuna, Naren Tallapragada, Adrian Veres, Victor Li, Leonid Peshkin, David A Weitz, and Marc W Kirschner. 2015. “Droplet Barcoding for Single-Cell Transcriptomics Applied to Embryonic Stem Cells.” <em>Cell</em> 161 (5): 1187–1201.</p>
</div>
<div id="ref-macosko2015highly">
<p>Macosko, Evan Z, Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. “Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.” <em>Cell</em> 161 (5): 1202–14.</p>
</div>
<div id="ref-picelli2013smart">
<p>Picelli, Simone, Åsa K Björklund, Omid R Faridani, Sven Sagasser, Gösta Winberg, and Rickard Sandberg. 2013. “Smart-Seq2 for Sensitive Full-Length Transcriptome Profiling in Single Cells.” <em>Nature Methods</em> 10 (11): 1096–8.</p>
</div>
<div id="ref-ramskold2012full">
<p>Ramsköld, Daniel, Shujun Luo, Yu-Chieh Wang, Robin Li, Qiaolin Deng, Omid R Faridani, Gregory A Daniels, et al. 2012. “Full-Length mRNA-Seq from Single-Cell Levels of Rna and Individual Circulating Tumor Cells.” <em>Nature Biotechnology</em> 30 (8): 777–82.</p>
</div>
<div id="ref-rosenberg2018single">
<p>Rosenberg, Alexander B, Charles M Roco, Richard A Muscat, Anna Kuchina, Paul Sample, Zizhen Yao, Lucas T Graybuck, et al. 2018. “Single-Cell Profiling of the Developing Mouse Brain and Spinal Cord with Split-Pool Barcoding.” <em>Science</em> 360 (6385): 176–82.</p>
</div>
<div id="ref-sasagawa2018quartz">
<p>Sasagawa, Yohei, Hiroki Danno, Hitomi Takada, Masashi Ebisawa, Kaori Tanaka, Tetsutaro Hayashi, Akira Kurisaki, and Itoshi Nikaido. 2018. “Quartz-Seq2: A High-Throughput Single-Cell Rna-Sequencing Method That Effectively Uses Limited Sequence Reads.” <em>Genome Biology</em> 19 (1): 1–24.</p>
</div>
<div id="ref-sasagawa2013quartz">
<p>Sasagawa, Yohei, Itoshi Nikaido, Tetsutaro Hayashi, Hiroki Danno, Kenichiro D Uno, Takeshi Imai, and Hiroki R Ueda. 2013. “Quartz-Seq: A Highly Reproducible and Sensitive Single-Cell Rna Sequencing Method, Reveals Non-Genetic Gene-Expression Heterogeneity.” <em>Genome Biology</em> 14 (4): 1–17.</p>
</div>
<div id="ref-sheng2017effective">
<p>Sheng, Kuanwei, Wenjian Cao, Yichi Niu, Qing Deng, and Chenghang Zong. 2017. “Effective Detection of Variation in Single-Cell Transcriptomes Using Matq-Seq.” <em>Nature Methods</em> 14 (3): 267–70.</p>
</div>
<div id="ref-tang2009mrna">
<p>Tang, Fuchou, Catalin Barbacioru, Yangzhou Wang, Ellen Nordman, Clarence Lee, Nanlan Xu, Xiaohui Wang, et al. 2009. “MRNA-Seq Whole-Transcriptome Analysis of a Single Cell.” <em>Nature Methods</em> 6 (5): 377–82.</p>
</div>
<div id="ref-tse_2018">
<p>Tse, David. 2018. “Lecture 16: Single Cell Rna-Seq - Introduction.” <em>Data Science for High-Throughput Sequencing</em>. <a href="http://data-science-sequencing.github.io/Win2018/lectures/lecture16/">http://data-science-sequencing.github.io/Win2018/lectures/lecture16/</a>.</p>
</div>
<div id="ref-zheng2017massively">
<p>Zheng, Grace XY, Jessica M Terry, Phillip Belgrader, Paul Ryvkin, Zachary W Bent, Ryan Wilson, Solongo B Ziraldo, et al. 2017. “Massively Parallel Digital Transcriptional Profiling of Single Cells.” <em>Nature Communications</em> 8 (1): 1–12.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>For more information: <a href="http://data-science-sequencing.github.io/Win2018/lectures/lecture16/" class="uri">http://data-science-sequencing.github.io/Win2018/lectures/lecture16/</a><a href="background.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="raw2matrix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
