<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Analysis Using Seurat | Fundamentals of scRNASeq Analysis</title>
  <meta name="description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Analysis Using Seurat | Fundamentals of scRNASeq Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="github-repo" content="linxy29/Fundamental-scRNA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Analysis Using Seurat | Fundamentals of scRNASeq Analysis" />
  
  <meta name="twitter:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  

<meta name="author" content="Xinyi Lin, Gordon Qian, Joshua Ho contact: jwkho@hku.hk" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="raw2matrix.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Fundamental scRNASeq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prerequisites</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-data"><i class="fa fa-check"></i>Download data</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Introduction to Single Cell Technology</a><ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#single-cell-rna-sequencing-scrna-seq"><i class="fa fa-check"></i><b>1.1</b> Single cell RNA sequencing (scRNA-seq)</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#cell-isolation"><i class="fa fa-check"></i><b>1.2</b> Cell isolation</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#barcodes-and-unique-molecular-identifiers-umi"><i class="fa fa-check"></i><b>1.3</b> Barcodes and Unique molecular identifiers (UMI)</a></li>
<li class="chapter" data-level="1.4" data-path="background.html"><a href="background.html#summary-of-widely-used-scrna-seq-technologies"><i class="fa fa-check"></i><b>1.4</b> Summary of widely used scRNA-seq technologies</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="raw2matrix.html"><a href="raw2matrix.html"><i class="fa fa-check"></i><b>2</b> Pre-processing and QC</a><ul>
<li class="chapter" data-level="2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#fastq-file"><i class="fa fa-check"></i><b>2.1</b> FASTQ file</a></li>
<li class="chapter" data-level="2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#cell-ranger"><i class="fa fa-check"></i><b>2.2</b> Cell Ranger</a><ul>
<li class="chapter" data-level="2.2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#general-workflow"><i class="fa fa-check"></i><b>2.2.1</b> General workflow</a></li>
<li class="chapter" data-level="2.2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#output-file"><i class="fa fa-check"></i><b>2.2.2</b> Output file</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="raw2matrix.html"><a href="raw2matrix.html#starsolo"><i class="fa fa-check"></i><b>2.3</b> STARsolo</a></li>
<li class="chapter" data-level="2.4" data-path="raw2matrix.html"><a href="raw2matrix.html#doublets"><i class="fa fa-check"></i><b>2.4</b> Doublets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downstream.html"><a href="downstream.html"><i class="fa fa-check"></i><b>3</b> Analysis Using Seurat</a><ul>
<li class="chapter" data-level="3.1" data-path="downstream.html"><a href="downstream.html#seurat-object"><i class="fa fa-check"></i><b>3.1</b> Seurat object</a></li>
<li class="chapter" data-level="3.2" data-path="downstream.html"><a href="downstream.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="downstream.html"><a href="downstream.html#standard-pre-processing-workflow"><i class="fa fa-check"></i><b>3.3</b> Standard pre-processing workflow</a><ul>
<li class="chapter" data-level="3.3.1" data-path="downstream.html"><a href="downstream.html#qc-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>3.3.1</b> QC and selecting cells for further analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="downstream.html"><a href="downstream.html#normalizing-the-data"><i class="fa fa-check"></i><b>3.3.2</b> Normalizing the data</a></li>
<li class="chapter" data-level="3.3.3" data-path="downstream.html"><a href="downstream.html#identification-of-highly-variable-features-feature-selection"><i class="fa fa-check"></i><b>3.3.3</b> Identification of highly variable features (feature selection)</a></li>
<li class="chapter" data-level="3.3.4" data-path="downstream.html"><a href="downstream.html#scaling-the-data"><i class="fa fa-check"></i><b>3.3.4</b> Scaling the data</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="downstream.html"><a href="downstream.html#perform-linear-dimensional-reduction"><i class="fa fa-check"></i><b>3.4</b> Perform linear dimensional reduction</a></li>
<li class="chapter" data-level="3.5" data-path="downstream.html"><a href="downstream.html#cluster-the-cells"><i class="fa fa-check"></i><b>3.5</b> Cluster the cells</a></li>
<li class="chapter" data-level="3.6" data-path="downstream.html"><a href="downstream.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>3.6</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fundamentals of scRNASeq Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downstream" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Analysis Using Seurat</h1>
<p>The contents in this chapter are adapted from <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">Seurat - Guided Clustering Tutorial</a> with little modification. The data we used is a 10k PBMC data getting from <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">10x Genomics website</a>.</p>
<p>In this tutorial, we will learn how to <code>Read 10X sequencing data and change it into a seurat object</code>, <code>QC and selecting cells for further analysis</code>, <code>Normalizing the data</code>, <code>Identification of highly variable features (feature selection)</code>, <code>Scaling the data</code>, <code>Perform linear dimensional reduction</code> and <code>Visualization</code>.</p>
<div id="seurat-object" class="section level2">
<h2><span class="header-section-number">3.1</span> Seurat object</h2>
<p>The Seurat object serves as a container that contains both data (like the count matrix) and analysis (like PCA, or clustering results) for a single-cell dataset. Before using Seurat to analyze scRNA-seq data, we can first have some basic understanding about the <code>Seurat object</code> from <a href="https://github.com/satijalab/seurat/wiki/Seurat#object-information">here</a>.</p>
</div>
<div id="setup-the-seurat-object" class="section level2">
<h2><span class="header-section-number">3.2</span> Setup the Seurat object</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="downstream.html#cb15-1"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb15-2"><a href="downstream.html#cb15-2"></a><span class="kw">library</span>(Seurat)</span>
<span id="cb15-3"><a href="downstream.html#cb15-3"></a><span class="kw">library</span>(patchwork)</span>
<span id="cb15-4"><a href="downstream.html#cb15-4"></a><span class="kw">library</span>(ggplot2)</span></code></pre></div>
<p>We start by reading in the data. The <code>Read10X_h5</code> reads count matrix from 10X CellRanger hdf5 file, returning a unique molecular identified (UMI) count matrix. The values in this matrix represent the number of molecules for each feature (i.e. gene; row) that are detected in each cell (column). It can be used to read both scATAC-seq and scRNA-seq matrices.</p>
<p>We next use the count matrix to create a <code>Seurat object</code>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="downstream.html#cb16-1"></a><span class="co"># Load the PBMC dataset</span></span>
<span id="cb16-2"><a href="downstream.html#cb16-2"></a>pbmc.data &lt;-<span class="st"> </span><span class="kw">Read10X_h5</span>(<span class="st">&quot;./data/10k_PBMC.h5&quot;</span>)</span>
<span id="cb16-3"><a href="downstream.html#cb16-3"></a><span class="co"># Initialize the Seurat object with the raw (non-normalized data).</span></span>
<span id="cb16-4"><a href="downstream.html#cb16-4"></a>pbmc &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> pbmc.data, <span class="dt">project =</span> <span class="st">&quot;pbmc10k&quot;</span>, <span class="dt">min.cells =</span> <span class="dv">3</span>, <span class="dt">min.features =</span> <span class="dv">200</span>)</span>
<span id="cb16-5"><a href="downstream.html#cb16-5"></a>pbmc</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 22432 features across 10813 samples within 1 assay 
## Active assay: RNA (22432 features, 0 variable features)</code></pre>
<p>If we want to read data using the output of the cellranger pipeline from 10X directly, we can use <code>Read10X()</code>.</p>
</div>
<div id="standard-pre-processing-workflow" class="section level2">
<h2><span class="header-section-number">3.3</span> Standard pre-processing workflow</h2>
<p>The steps below encompass the standard pre-processing workflow for scRNA-seq data in Seurat. They are based on the RNA reads count matrix we will get from <code>Cell Ranger</code> or <code>STARsolo</code> output. The standard pre-processing workflow represents the selection and filtration of cells based on QC metrics, data normalization and scaling, and the detection of highly variable features.</p>
<div id="qc-and-selecting-cells-for-further-analysis" class="section level3">
<h3><span class="header-section-number">3.3.1</span> QC and selecting cells for further analysis</h3>
<p>Seurat allows you to easily explore QC metrics and filter cells based on any user-defined criteria. A few QC metrics commonly used <span class="citation">(Ilicic et al. <a href="#ref-ilicic2016classification" role="doc-biblioref">2016</a>)</span> by the community include</p>
<ul>
<li>The number of unique genes detected in each cell.
<ul>
<li>low-quality cells or empty droplets will often have very few genes</li>
<li>cell doublets or multiplets may exhibit an aberrantly high gene count</li>
<li>similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)</li>
</ul></li>
<li>The percentage of reads that map to the mitochondrial genome
<ul>
<li>low-quality / dying cells often exhibit extensive mitochondrial contamination</li>
<li>we calculate mitochondrial QC metrics with the <code>PercentageFeatureSet()</code> function, which calculates the percentage of counts originating from a set of features</li>
<li>we use the set of all genes starting with MT- as a set of mitochondrial genes</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="downstream.html#cb18-1"></a><span class="co"># The [[ operator can add columns to object metadata. This is a great place to stash QC stats</span></span>
<span id="cb18-2"><a href="downstream.html#cb18-2"></a>pbmc[[<span class="st">&quot;percent.mt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(pbmc, <span class="dt">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span></code></pre></div>
<p>The number of unique genes and total molecules are automatically calculated during <code>CreateSeuratObject()</code>. They are stored in the object meta data.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="downstream.html#cb19-1"></a><span class="co"># Show QC metrics for the first 5 cells in the control group</span></span>
<span id="cb19-2"><a href="downstream.html#cb19-2"></a><span class="kw">head</span>(pbmc<span class="op">@</span>meta.data, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>##                    orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCCAGTATATGGA-1    pbmc10k        860          350 44.1860465
## AAACCCAGTATCGTAC-1    pbmc10k       1548          729  0.4521964
## AAACCCAGTCGGTGAA-1    pbmc10k       6387         1827 10.4117739
## AAACCCAGTTAGAAAC-1    pbmc10k      16664         3744  5.2808449
## AAACCCAGTTATCTTC-1    pbmc10k       3352         1464 13.8424821</code></pre>
<p>We can visualize the <code>nFeature_RNA</code>, <code>nCount_RNA</code> and <code>percent.mt</code> we used as QC metrics.</p>
<p>* Just like <code>Cell Ranger</code> output, <code>feature</code> in the following results represents gene. <code>nFeature_RNA</code> is the number of genes detected in each cell. <code>nCount_RNA</code> is the total number of molecules detected within a cell. And each dot in the following plots represents a cell.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="downstream.html#cb21-1"></a><span class="co"># Visualize QC metrics as a violin plot</span></span>
<span id="cb21-2"><a href="downstream.html#cb21-2"></a><span class="kw">VlnPlot</span>(pbmc, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">pt.size =</span> <span class="fl">0.0001</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p>We can then using dot plots to show the relationship between <code>nCount_RNA</code>, <code>nFeature_RNA</code> and <code>percent.mt</code>.</p>
<p>* The number <code>-0.35</code> and <code>0.94</code> above each plot denotes the correlations between x-axis and y-axis.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="downstream.html#cb22-1"></a><span class="co"># FeatureScatter is typically used to visualize feature-feature relationships, but can be used</span></span>
<span id="cb22-2"><a href="downstream.html#cb22-2"></a><span class="co"># for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</span></span>
<span id="cb22-3"><a href="downstream.html#cb22-3"></a>plot1 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(pbmc, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb22-4"><a href="downstream.html#cb22-4"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb22-5"><a href="downstream.html#cb22-5"></a>plot2 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(pbmc, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb22-6"><a href="downstream.html#cb22-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb22-7"><a href="downstream.html#cb22-7"></a>plot1 <span class="op">+</span><span class="st"> </span>plot2</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<p>Here, we filter away cells that have unique feature counts(genes) over 5,000 or less than 200. We also filter away cells that have &gt; 15% mitochondrial counts</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="downstream.html#cb23-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">subset</span>(pbmc, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span><span class="dv">200</span> <span class="op">&amp;</span><span class="st"> </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span><span class="dv">5000</span> <span class="op">&amp;</span><span class="st"> </span>percent.mt <span class="op">&lt;</span><span class="st"> </span><span class="dv">15</span>)</span></code></pre></div>
<p>We can visualize QC metrics again after filtering cells</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="downstream.html#cb24-1"></a><span class="kw">VlnPlot</span>(pbmc, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">pt.size =</span> <span class="fl">0.001</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="downstream.html#cb25-1"></a>plot1 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(pbmc, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-2"><a href="downstream.html#cb25-2"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb25-3"><a href="downstream.html#cb25-3"></a>plot2 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(pbmc, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-4"><a href="downstream.html#cb25-4"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb25-5"><a href="downstream.html#cb25-5"></a>plot1 <span class="op">+</span><span class="st"> </span>plot2</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-32-2.png" width="672" /></p>
<p>* Depend on the data we analyze, we can use different cutoff for <code>nFeature_RNA</code> and <code>percent.mt</code>. For example, we can filter away cells that have unique feature counts(genes) over 5,000 or less than 300, or cells that have &gt; 10% mitochondrial counts, and see how QC metrics looks like.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="downstream.html#cb26-1"></a>temp &lt;-<span class="st"> </span><span class="kw">subset</span>(pbmc, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span><span class="dv">300</span> <span class="op">&amp;</span><span class="st"> </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span><span class="dv">5000</span> <span class="op">&amp;</span><span class="st"> </span>percent.mt <span class="op">&lt;</span><span class="st"> </span><span class="dv">10</span>)</span>
<span id="cb26-2"><a href="downstream.html#cb26-2"></a><span class="kw">VlnPlot</span>(temp, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">pt.size =</span> <span class="fl">0.001</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="downstream.html#cb27-1"></a>plot1 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(temp, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb27-2"><a href="downstream.html#cb27-2"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb27-3"><a href="downstream.html#cb27-3"></a>plot2 &lt;-<span class="st"> </span><span class="kw">FeatureScatter</span>(temp, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb27-4"><a href="downstream.html#cb27-4"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb27-5"><a href="downstream.html#cb27-5"></a>plot1 <span class="op">+</span><span class="st"> </span>plot2</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-33-2.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="downstream.html#cb28-1"></a><span class="kw">rm</span>(temp)</span></code></pre></div>
</div>
<div id="normalizing-the-data" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Normalizing the data</h3>
<p>After removing unwanted cells from the dataset, the next step is to normalize the data. By default, we employ a global-scaling normalization method “LogNormalize” that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result. Normalized values are stored in <code>pbmc[["RNA"]]@data</code>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="downstream.html#cb29-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(pbmc, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="dv">10000</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>Here we sample 10,000 reads counts from the large gene expression matrix to visualize the gene expression distribution before and after normalization separately (zeros are not included).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="downstream.html#cb30-1"></a><span class="co"># set seed and put two plots in one figure</span></span>
<span id="cb30-2"><a href="downstream.html#cb30-2"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb30-3"><a href="downstream.html#cb30-3"></a><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb30-4"><a href="downstream.html#cb30-4"></a><span class="co"># original expression distribution</span></span>
<span id="cb30-5"><a href="downstream.html#cb30-5"></a>raw_geneExp =<span class="st"> </span><span class="kw">as.vector</span>(pbmc[[<span class="st">&#39;RNA&#39;</span>]]<span class="op">@</span>counts) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample</span>(<span class="dv">10000</span>)</span>
<span id="cb30-6"><a href="downstream.html#cb30-6"></a>raw_geneExp =<span class="st"> </span>raw_geneExp[raw_geneExp <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb30-7"><a href="downstream.html#cb30-7"></a><span class="kw">hist</span>(raw_geneExp)</span>
<span id="cb30-8"><a href="downstream.html#cb30-8"></a><span class="co"># expression distribution after normalization</span></span>
<span id="cb30-9"><a href="downstream.html#cb30-9"></a>logNorm_geneExp =<span class="st"> </span><span class="kw">as.vector</span>(pbmc[[<span class="st">&#39;RNA&#39;</span>]]<span class="op">@</span>data) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sample</span>(<span class="dv">10000</span>)</span>
<span id="cb30-10"><a href="downstream.html#cb30-10"></a>logNorm_geneExp =<span class="st"> </span>logNorm_geneExp[logNorm_geneExp <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>]</span>
<span id="cb30-11"><a href="downstream.html#cb30-11"></a><span class="kw">hist</span>(logNorm_geneExp)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
<p>In this previous line of code (and in future commands), <code>Seurat</code> provides the default values for certain parameters in the function call. However, this isn’t required and the same behavior can be achieved with:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="downstream.html#cb31-1"></a><span class="co"># same as the above command, skip here</span></span>
<span id="cb31-2"><a href="downstream.html#cb31-2"></a>pbmc &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(pbmc)</span></code></pre></div>
</div>
<div id="identification-of-highly-variable-features-feature-selection" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Identification of highly variable features (feature selection)</h3>
<p>We next calculate a subset of features that exhibit high cell-to-cell variation in the dataset (i.e, they are highly expressed in some cells, and lowly expressed in others). It is shown that <span class="citation">(Brennecke et al. <a href="#ref-brennecke2013accounting" role="doc-biblioref">2013</a>)</span> focusing on these genes in downstream analysis helps to highlight biological signal in single-cell datasets.</p>
<p>The procedure in Seurat is described in detail here <span class="citation">(Stuart et al. <a href="#ref-stuart2019comprehensive" role="doc-biblioref">2019</a>)</span>, and improves on previous versions by directly modeling the mean-variance relationship inherent in single-cell data, and is implemented in the <code>FindVariableFeatures()</code> function. By default, <code>Seurat</code> returns 2,000 features per dataset and these will be used in downstream analysis, like PCA.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="downstream.html#cb32-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">FindVariableFeatures</span>(pbmc, <span class="dt">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="dt">nfeatures =</span> <span class="dv">2000</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb32-2"><a href="downstream.html#cb32-2"></a></span>
<span id="cb32-3"><a href="downstream.html#cb32-3"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb32-4"><a href="downstream.html#cb32-4"></a>top10 &lt;-<span class="st"> </span><span class="kw">head</span>(<span class="kw">VariableFeatures</span>(pbmc), <span class="dv">10</span>)</span>
<span id="cb32-5"><a href="downstream.html#cb32-5"></a></span>
<span id="cb32-6"><a href="downstream.html#cb32-6"></a><span class="co"># plot variable features with and without labels</span></span>
<span id="cb32-7"><a href="downstream.html#cb32-7"></a>plot1 &lt;-<span class="st"> </span><span class="kw">VariableFeaturePlot</span>(pbmc) <span class="op">+</span><span class="st"> </span></span>
<span id="cb32-8"><a href="downstream.html#cb32-8"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;top&quot;</span>)</span>
<span id="cb32-9"><a href="downstream.html#cb32-9"></a>plot2 &lt;-<span class="st"> </span><span class="kw">LabelPoints</span>(<span class="dt">plot =</span> plot1, <span class="dt">points =</span> top10, <span class="dt">repel =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb32-10"><a href="downstream.html#cb32-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb32-11"><a href="downstream.html#cb32-11"></a>plot1 <span class="op">+</span><span class="st"> </span>plot2</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
</div>
<div id="scaling-the-data" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Scaling the data</h3>
<p>Next, we apply a linear transformation (‘scaling’) that is a standard pre-processing step prior to dimensional reduction techniques like PCA. The <code>ScaleData()</code> function:</p>
<ul>
<li>shifts the expression of each gene, so that the mean expression across cells is 0</li>
<li>scales the expression of each gene, so that the variance across cells is 1
<ul>
<li>this step gives equal weight in downstream analyses, so that highly-expressed genes do not dominate</li>
</ul></li>
<li>the results of this are stored in <code>pbmc[["RNA"]]@scale.data</code></li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="downstream.html#cb33-1"></a>all.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(pbmc)</span>
<span id="cb33-2"><a href="downstream.html#cb33-2"></a>pbmc &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(pbmc, <span class="dt">features =</span> all.genes, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>When using the above command, we use all genes to scale data. Scaling is an essential step in the Seurat workflow, but only on genes that will be used as input to PCA. Therefore, the default in <code>ScaleData()</code> is only to perform scaling on the previously identified variable features (2,000 by default). And it will make this step faster.</p>
<p>In this case, our PCA and clustering results will be unaffected. However, Seurat heatmaps (produced as shown below with <code>DoHeatmap()</code>) require genes in the heatmap to be scaled, to make sure highly-expressed genes don’t dominate the heatmap. To make sure we don’t leave any genes out of the heatmap later, we are scaling all genes in this tutorial.</p>
<p>If we want to scale on the previously identified 2000 variable features, we can simply omit the features argument in the previous function call, i.e.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="downstream.html#cb34-1"></a><span class="co"># use the above one, skip here</span></span>
<span id="cb34-2"><a href="downstream.html#cb34-2"></a>pbmc &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(pbmc)</span></code></pre></div>
<p>The <code>ScaleData()</code> function also allow us to remove unwanted sources of variation from a single-cell dataset. For example, we could ‘regress out’ heterogeneity associated with (for example) cell cycle stage, or mitochondrial contamination. This feature can be achieved by specifying <code>vars.to.regress</code>, i.e.:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="downstream.html#cb35-1"></a><span class="co"># skip here</span></span>
<span id="cb35-2"><a href="downstream.html#cb35-2"></a>pbmc &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(pbmc, <span class="dt">vars.to.regress =</span> <span class="st">&quot;percent.mt&quot;</span>)</span></code></pre></div>
<p>However, particularly for advanced users who would like to use this functionality, it is recommended by <code>Seurat</code> using their new normalization workflow, <code>SCTransform()</code>. The method is described in Seurat paper <span class="citation">(Hafemeister and Satija <a href="#ref-hafemeister2019normalization" role="doc-biblioref">2019</a>)</span>, with a separate vignette using Seurat v3 <a href="https://satijalab.org/seurat/articles/sctransform_vignette.html">here</a>. As with <code>ScaleData()</code>, the function <code>SCTransform()</code> also includes a <code>vars.to.regress</code> parameter.</p>
</div>
</div>
<div id="perform-linear-dimensional-reduction" class="section level2">
<h2><span class="header-section-number">3.4</span> Perform linear dimensional reduction</h2>
<p>Next we perform PCA on the scaled data. By default, only the previously determined variable features are used as input, but can be defined using <code>features</code> argument if you wish to choose a different subset.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="downstream.html#cb36-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(pbmc, <span class="dt">features =</span> <span class="kw">VariableFeatures</span>(<span class="dt">object =</span> pbmc), <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>Seurat provides several useful ways of visualizing both cells and features that define the PCA, including <code>VizDimLoadings()</code>, <code>DimPlot()</code>, and <code>DimHeatmap()</code>.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="downstream.html#cb37-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb37-2"><a href="downstream.html#cb37-2"></a><span class="kw">print</span>(pbmc[[<span class="st">&quot;pca&quot;</span>]], <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">nfeatures =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  LTB, IL32, TRAC, CD3D, TRBC2 
## Negative:  FCN1, FGL2, CST3, IFI30, TYMP 
## PC_ 2 
## Positive:  MS4A1, CD79A, BANK1, IGHM, NIBAN3 
## Negative:  IL32, GZMM, CD3D, CD7, CD247 
## PC_ 3 
## Positive:  GZMB, CLIC3, NKG7, GNLY, KLRD1 
## Negative:  CCR7, LEF1, TRABD2A, TCF7, LTB 
## PC_ 4 
## Positive:  CD79B, MS4A1, GNLY, CD79A, LINC00926 
## Negative:  LILRA4, CLEC4C, SERPINF1, TPM2, SCT 
## PC_ 5 
## Positive:  CDKN1C, HES4, CTSL, TCF7L2, BATF3 
## Negative:  S100A12, ITGAM, VCAN, CES1, MGST1</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="downstream.html#cb39-1"></a><span class="kw">VizDimLoadings</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="downstream.html#cb40-1"></a><span class="kw">DimPlot</span>(pbmc, <span class="dt">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-42-2.png" width="672" /></p>
<p>In particular <code>DimHeatmap()</code> allows for easy exploration of the primary sources of heterogeneity in a dataset, and can be useful when trying to decide which PCs to include for further downstream analyses. Both cells and features are ordered according to their PCA scores. Setting <code>cells</code> to a number plots the ‘extreme’ cells on both ends of the spectrum, which dramatically speeds plotting for large datasets. Though clearly a supervised analysis, we find this to be a valuable tool for exploring correlated feature sets.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="downstream.html#cb41-1"></a><span class="kw">DimHeatmap</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span>, <span class="dt">cells =</span> <span class="dv">500</span>, <span class="dt">balanced =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="downstream.html#cb42-1"></a><span class="kw">DimHeatmap</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">9</span>, <span class="dt">cells =</span> <span class="dv">500</span>, <span class="dt">balanced =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-43-2.png" width="672" /></p>
<p>To overcome the extensive technical noise in any single feature for scRNA-seq data, Seurat clusters cells based on their PCA scores, with each PC essentially representing a ‘metafeature’ that combines information across a correlated feature set. The top principal components therefore represent a robust compression of the dataset. Here, we choose first <strong>20 PCs</strong>.</p>
</div>
<div id="cluster-the-cells" class="section level2">
<h2><span class="header-section-number">3.5</span> Cluster the cells</h2>
<p><code>Seurat v3</code> applies a graph-based clustering approach, building upon initial strategies in <span class="citation">(Macosko et al. <a href="#ref-macosko2015highly" role="doc-biblioref">2015</a>)</span>. Importantly, the <em>distance metric</em> which drives the clustering analysis (based on previously identified PCs) remains the same. However, approach in <code>Seurat</code> to partitioning the cellular distance matrix into clusters has dramatically improved. This approach was heavily inspired by recent publications which applied graph-based clustering approaches to scRNA-seq data <span class="citation">(Xu and Su <a href="#ref-xu2015identification" role="doc-biblioref">2015</a>)</span> and CyTOF data <span class="citation">(Levine et al. <a href="#ref-levine2015data" role="doc-biblioref">2015</a>)</span>. Briefly, these methods embed cells in a graph structure - for example a K-nearest neighbor (KNN) graph, with edges drawn between cells with similar feature expression patterns, and then attempt to partition this graph into highly interconnected ‘quasi-cliques’ or ‘communities’.</p>
<p>As in PhenoGraph, they first construct a KNN graph based on the euclidean distance in PCA space, and refine the edge weights between any two cells based on the shared overlap in their local neighborhoods (Jaccard similarity). This step is performed using the <code>FindNeighbors()</code> function, and takes as input the previously defined dimensionality of the dataset (first 20 PCs).</p>
<p>To cluster the cells, we next apply modularity optimization techniques such as the Louvain algorithm (default) or SLM <span class="citation">(Blondel et al. <a href="#ref-blondel2008fast" role="doc-biblioref">2008</a>)</span>, to iteratively group cells together, with the goal of optimizing the standard modularity function. The <code>FindClusters()</code> function implements this procedure, and contains a resolution parameter that sets the ‘granularity’ of the downstream clustering, with increased values leading to a greater number of clusters. We find that setting this parameter between 0.4-1.2 typically returns good results for single-cell datasets of around 3K cells. Optimal resolution often increases for larger datasets. The clusters can be found using the <code>Idents()</code> function.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="downstream.html#cb43-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb43-2"><a href="downstream.html#cb43-2"></a>pbmc &lt;-<span class="st"> </span><span class="kw">FindClusters</span>(pbmc, <span class="dt">resolution =</span> <span class="fl">0.5</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="downstream.html#cb44-1"></a><span class="co"># Look at cluster IDs of the first 5 cells</span></span>
<span id="cb44-2"><a href="downstream.html#cb44-2"></a><span class="kw">head</span>(<span class="kw">Idents</span>(pbmc), <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## AAACCCAGTATCGTAC-1 AAACCCAGTCGGTGAA-1 AAACCCAGTTAGAAAC-1 AAACCCAGTTATCTTC-1 
##                  2                  4                  2                  8 
## AAACCCAGTTGCCGAC-1 
##                  0 
## Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15</code></pre>
</div>
<div id="run-non-linear-dimensional-reduction-umaptsne" class="section level2">
<h2><span class="header-section-number">3.6</span> Run non-linear dimensional reduction (UMAP/tSNE)</h2>
<p>Seurat offers several non-linear dimensional reduction techniques, such as tSNE and UMAP, to visualize and explore these datasets. The goal of these algorithms is to learn the underlying manifold of the data in order to place similar cells together in low-dimensional space. Cells within the graph-based clusters determined above should co-localize on these dimension reduction plots. As input to the UMAP and tSNE, we suggest using the same PCs as input to the clustering analysis.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="downstream.html#cb46-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
## To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
## This message will be shown once per session</code></pre>
<p>Then we can get the UMAP plot of the single cell clustering results.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="downstream.html#cb48-1"></a><span class="kw">DimPlot</span>(pbmc, <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-47-1.png" width="672" /></p>
<p>We can also visualize it using tSNE plot</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="downstream.html#cb49-1"></a>pbmc &lt;-<span class="st"> </span><span class="kw">RunTSNE</span>(pbmc, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb49-2"><a href="downstream.html#cb49-2"></a><span class="kw">DimPlot</span>(pbmc, <span class="dt">reduction =</span> <span class="st">&quot;tsne&quot;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
<p>We can set <code>label = TRUE</code> or use the <code>LabelClusters</code> function to help label individual clusters.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="downstream.html#cb50-1"></a><span class="kw">DimPlot</span>(pbmc, <span class="dt">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="dt">label =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="downstream.html#cb51-1"></a>plot &lt;-<span class="st"> </span><span class="kw">DimPlot</span>(<span class="dt">object =</span> pbmc)</span>
<span id="cb51-2"><a href="downstream.html#cb51-2"></a><span class="kw">LabelClusters</span>(<span class="dt">plot =</span> plot, <span class="dt">id =</span> <span class="st">&#39;ident&#39;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-49-2.png" width="672" /></p>
<p>Now it is time to save the object so that it can easily be loaded back in without having to rerun the computationally intensive steps performed above, or easily shared with collaborators.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="downstream.html#cb52-1"></a><span class="kw">saveRDS</span>(pbmc, <span class="dt">file =</span> <span class="st">&quot;./data/pbmc_processed.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-blondel2008fast">
<p>Blondel, Vincent D, Jean-Loup Guillaume, Renaud Lambiotte, and Etienne Lefebvre. 2008. “Fast Unfolding of Communities in Large Networks.” <em>Journal of Statistical Mechanics: Theory and Experiment</em> 2008 (10): P10008.</p>
</div>
<div id="ref-brennecke2013accounting">
<p>Brennecke, Philip, Simon Anders, Jong Kyoung Kim, Aleksandra A Kołodziejczyk, Xiuwei Zhang, Valentina Proserpio, Bianka Baying, et al. 2013. “Accounting for Technical Noise in Single-Cell Rna-Seq Experiments.” <em>Nature Methods</em> 10 (11): 1093–5.</p>
</div>
<div id="ref-hafemeister2019normalization">
<p>Hafemeister, Christoph, and Rahul Satija. 2019. “Normalization and Variance Stabilization of Single-Cell Rna-Seq Data Using Regularized Negative Binomial Regression.” <em>Genome Biology</em> 20 (1): 1–15.</p>
</div>
<div id="ref-ilicic2016classification">
<p>Ilicic, Tomislav, Jong Kyoung Kim, Aleksandra A Kolodziejczyk, Frederik Otzen Bagger, Davis James McCarthy, John C Marioni, and Sarah A Teichmann. 2016. “Classification of Low Quality Cells from Single-Cell Rna-Seq Data.” <em>Genome Biology</em> 17 (1): 1–15.</p>
</div>
<div id="ref-levine2015data">
<p>Levine, Jacob H, Erin F Simonds, Sean C Bendall, Kara L Davis, D Amir El-ad, Michelle D Tadmor, Oren Litvin, et al. 2015. “Data-Driven Phenotypic Dissection of Aml Reveals Progenitor-Like Cells That Correlate with Prognosis.” <em>Cell</em> 162 (1): 184–97.</p>
</div>
<div id="ref-macosko2015highly">
<p>Macosko, Evan Z, Anindita Basu, Rahul Satija, James Nemesh, Karthik Shekhar, Melissa Goldman, Itay Tirosh, et al. 2015. “Highly Parallel Genome-Wide Expression Profiling of Individual Cells Using Nanoliter Droplets.” <em>Cell</em> 161 (5): 1202–14.</p>
</div>
<div id="ref-stuart2019comprehensive">
<p>Stuart, Tim, Andrew Butler, Paul Hoffman, Christoph Hafemeister, Efthymia Papalexi, William M Mauck III, Yuhan Hao, Marlon Stoeckius, Peter Smibert, and Rahul Satija. 2019. “Comprehensive Integration of Single-Cell Data.” <em>Cell</em> 177 (7): 1888–1902.</p>
</div>
<div id="ref-xu2015identification">
<p>Xu, Chen, and Zhengchang Su. 2015. “Identification of Cell Types from Single-Cell Transcriptomes Using a Novel Clustering Method.” <em>Bioinformatics</em> 31 (12): 1974–80.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="raw2matrix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
