<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Downstream Analysis Using Seurat | Fundamentals of scRNASeq Analysis</title>
  <meta name="description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Downstream Analysis Using Seurat | Fundamentals of scRNASeq Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="github-repo" content="linxy29/Fundamental-scRNA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Downstream Analysis Using Seurat | Fundamentals of scRNASeq Analysis" />
  
  <meta name="twitter:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  

<meta name="author" content="Xinyi Lin, Gordon Qian, Joshua Ho" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="raw2matrix.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Fundamental scRNASeq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prerequisites</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-data"><i class="fa fa-check"></i>Download data</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Background Knowledge</a><ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#single-cell-rna-sequencingscrna-seq"><i class="fa fa-check"></i><b>1.1</b> Single cell RNA sequencing(scRNA-seq)</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#cell-isolation"><i class="fa fa-check"></i><b>1.2</b> Cell isolation</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#barcodes-and-unique-molecular-identifiersumi"><i class="fa fa-check"></i><b>1.3</b> Barcodes and Unique molecular identifiers(UMI)</a></li>
<li class="chapter" data-level="1.4" data-path="background.html"><a href="background.html#summary-of-widely-used-scrna-seq-technologies"><i class="fa fa-check"></i><b>1.4</b> Summary of widely used scRNA-seq technologies</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="raw2matrix.html"><a href="raw2matrix.html"><i class="fa fa-check"></i><b>2</b> Pre-processing 10X RNA-seq Data</a><ul>
<li class="chapter" data-level="2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#fastq-file"><i class="fa fa-check"></i><b>2.1</b> FASTQ file</a></li>
<li class="chapter" data-level="2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#cell-ranger"><i class="fa fa-check"></i><b>2.2</b> Cell Ranger</a><ul>
<li class="chapter" data-level="2.2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#general-workflow"><i class="fa fa-check"></i><b>2.2.1</b> General workflow</a></li>
<li class="chapter" data-level="2.2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#output-file"><i class="fa fa-check"></i><b>2.2.2</b> Output file</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="raw2matrix.html"><a href="raw2matrix.html#starsolo"><i class="fa fa-check"></i><b>2.3</b> STARsolo</a></li>
<li class="chapter" data-level="2.4" data-path="raw2matrix.html"><a href="raw2matrix.html#doublets"><i class="fa fa-check"></i><b>2.4</b> Doublets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downstream.html"><a href="downstream.html"><i class="fa fa-check"></i><b>3</b> Downstream Analysis Using Seurat</a><ul>
<li class="chapter" data-level="3.1" data-path="downstream.html"><a href="downstream.html#seurat-object"><i class="fa fa-check"></i><b>3.1</b> Seurat object</a></li>
<li class="chapter" data-level="3.2" data-path="downstream.html"><a href="downstream.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="downstream.html"><a href="downstream.html#standard-pre-processing-workflow"><i class="fa fa-check"></i><b>3.3</b> Standard pre-processing workflow</a><ul>
<li class="chapter" data-level="3.3.1" data-path="downstream.html"><a href="downstream.html#qc-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>3.3.1</b> QC and selecting cells for further analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="downstream.html"><a href="downstream.html#normalizing-the-data"><i class="fa fa-check"></i><b>3.3.2</b> Normalizing the data</a></li>
<li class="chapter" data-level="3.3.3" data-path="downstream.html"><a href="downstream.html#identification-of-highly-variable-features-feature-selection"><i class="fa fa-check"></i><b>3.3.3</b> Identification of highly variable features (feature selection)</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="downstream.html"><a href="downstream.html#perform-integration"><i class="fa fa-check"></i><b>3.4</b> Perform integration</a></li>
<li class="chapter" data-level="3.5" data-path="downstream.html"><a href="downstream.html#scaling-the-data"><i class="fa fa-check"></i><b>3.5</b> Scaling the data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fundamentals of scRNASeq Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downstream" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Downstream Analysis Using Seurat</h1>
<p>In this tutorial, we will learn how to <code>Read 10X sequencing data and change it into a seurat object</code>, <code>Standard pre-processing workflow</code>, <code>QC and selecting cells for further analysis</code>, <code>Create an ‘integrated’ data assay from data coming from two samples</code>, <code>Normalizing the data</code>, <code>Identification of highly variable features (feature selection)</code>, <code>Scaling the data</code>, <code>Perform linear dimensional reduction</code> and <code>Cluster the cells</code>.</p>
<p>Data used here are samples of human immune cells (PBMC) in either <a href="https://www.nature.com/articles/nbt.4042">a resting or interferon-stimulated state</a>.</p>
<div id="seurat-object" class="section level2">
<h2><span class="header-section-number">3.1</span> Seurat object</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="downstream.html#cb1-1"></a><span class="kw">library</span>(Seurat)</span>
<span id="cb1-2"><a href="downstream.html#cb1-2"></a><span class="kw">library</span>(SeuratData)</span>
<span id="cb1-3"><a href="downstream.html#cb1-3"></a><span class="kw">library</span>(patchwork)</span>
<span id="cb1-4"><a href="downstream.html#cb1-4"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="setup-the-seurat-object" class="section level2">
<h2><span class="header-section-number">3.2</span> Setup the Seurat object</h2>
<p>We start by reading in the data. The <code>Read10X()</code> function reads in the output of the cellranger pipeline from 10X, returning a unique molecular identified (UMI) count matrix. The values in this matrix represent the number of molecules for each feature (i.e. gene; row) that are detected in each cell (column).</p>
<p>We next use the count matrix to create a <code>Seurat</code> object. The object serves as a container that contains both data (like the count matrix) and analysis (like PCA, or clustering results) for a single-cell dataset. For a technical discussion of the Seurat object structure, check out our <a href="https://github.com/satijalab/seurat/wiki">GitHub Wiki</a>.</p>
<p>The sameple in <code>GSM2560248</code> is the control group which is human immune cells (PBMC) in a resting state, while the sameple in <code>GSM2560249</code> is the human immune cells (PBMC) in a interferon-stimulated state.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="downstream.html#cb2-1"></a>data_dir1 =<span class="st"> &quot;./data/Exp6_Kang/GSM2560248&quot;</span></span>
<span id="cb2-2"><a href="downstream.html#cb2-2"></a>expression_matrix1 &lt;-<span class="st"> </span><span class="kw">Read10X</span>(<span class="dt">data.dir =</span> data_dir1)</span>
<span id="cb2-3"><a href="downstream.html#cb2-3"></a>ctrl =<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> expression_matrix1)</span>
<span id="cb2-4"><a href="downstream.html#cb2-4"></a>ctrl =<span class="st"> </span><span class="kw">AddMetaData</span>(<span class="dt">object =</span> ctrl, <span class="dt">metadata =</span> <span class="kw">rep</span>(<span class="st">&quot;ctrl&quot;</span>, <span class="kw">dim</span>(ctrl)[<span class="dv">2</span>]), <span class="dt">col.name =</span> <span class="st">&#39;sample&#39;</span>)</span>
<span id="cb2-5"><a href="downstream.html#cb2-5"></a></span>
<span id="cb2-6"><a href="downstream.html#cb2-6"></a>data_dir2 =<span class="st"> &quot;./data/Exp6_Kang/GSM2560249&quot;</span></span>
<span id="cb2-7"><a href="downstream.html#cb2-7"></a>expression_matrix2 &lt;-<span class="st"> </span><span class="kw">Read10X</span>(<span class="dt">data.dir =</span> data_dir2)</span>
<span id="cb2-8"><a href="downstream.html#cb2-8"></a>stim =<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> expression_matrix2)</span>
<span id="cb2-9"><a href="downstream.html#cb2-9"></a>stim =<span class="st"> </span><span class="kw">AddMetaData</span>(<span class="dt">object =</span> stim, <span class="dt">metadata =</span> <span class="kw">rep</span>(<span class="st">&quot;stim&quot;</span>, <span class="kw">dim</span>(stim)[<span class="dv">2</span>]), <span class="dt">col.name =</span> <span class="st">&#39;sample&#39;</span>)</span></code></pre></div>
</div>
<div id="standard-pre-processing-workflow" class="section level2">
<h2><span class="header-section-number">3.3</span> Standard pre-processing workflow</h2>
<p>The steps below encompass the standard pre-processing workflow for scRNA-seq data in Seurat. These represent the selection and filtration of cells based on QC metrics, data normalization and scaling, and the detection of highly variable features.</p>
<div id="qc-and-selecting-cells-for-further-analysis" class="section level3">
<h3><span class="header-section-number">3.3.1</span> QC and selecting cells for further analysis</h3>
<p>Seurat allows you to easily explore QC metrics and filter cells based on any user-defined criteria. A few QC metrics commonly used by the community include</p>
<ul>
<li>The number of unique genes detected in each cell.
<ul>
<li>Low-quality cells or empty droplets will often have very few genes</li>
<li>Cell doublets or multiplets may exhibit an aberrantly high gene count</li>
<li>Similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)</li>
</ul></li>
<li>The percentage of reads that map to the mitochondrial genome
<ul>
<li>Low-quality / dying cells often exhibit extensive mitochondrial contamination</li>
<li>We calculate mitochondrial QC metrics with the PercentageFeatureSet() function, which calculates the percentage of counts originating from a set of features</li>
<li>We use the set of all genes starting with MT- as a set of mitochondrial genes</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="downstream.html#cb3-1"></a><span class="co"># The [[ operator can add columns to object metadata. This is a great place to stash QC stats</span></span>
<span id="cb3-2"><a href="downstream.html#cb3-2"></a>ctrl[[<span class="st">&quot;percent.mt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(ctrl, <span class="dt">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb3-3"><a href="downstream.html#cb3-3"></a>stim[[<span class="st">&quot;percent.mt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(stim, <span class="dt">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span></code></pre></div>
<p>The number of unique genes and total molecules are automatically calculated during <code>CreateSeuratObject()</code>. They are stored in the object meta data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="downstream.html#cb4-1"></a><span class="co"># Show QC metrics for the first 5 cells in the control group</span></span>
<span id="cb4-2"><a href="downstream.html#cb4-2"></a><span class="kw">head</span>(ctrl<span class="op">@</span>meta.data, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>##                     orig.ident nCount_RNA nFeature_RNA sample percent.mt
## AAACATACAATGCC-1 SeuratProject       2191          852   ctrl          0
## AAACATACATTTCC-1 SeuratProject       3018          878   ctrl          0
## AAACATACCAGAAA-1 SeuratProject       2481          713   ctrl          0
## AAACATACCAGCTA-1 SeuratProject       3157          950   ctrl          0
## AAACATACCATGCA-1 SeuratProject        703          337   ctrl          0</code></pre>
<p>We can visualize the <code>nFeature_RNA</code>, <code>nCount_RNA</code> and <code>percent.mt</code> we used for QC metrics</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="downstream.html#cb6-1"></a><span class="co"># Visualize QC metrics as a violin plot</span></span>
<span id="cb6-2"><a href="downstream.html#cb6-2"></a><span class="kw">VlnPlot</span>(ctrl, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">pt.size =</span> <span class="fl">0.00001</span>)</span></code></pre></div>
<pre><code>## Warning in SingleExIPlot(type = type, data = data[, x, drop = FALSE], idents =
## idents, : All cells have the same value of percent.mt.</code></pre>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>In this data, we the reads counts of mitochondrial genes are zero, thus the <code>percent.mt</code> is zero for all cells.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="downstream.html#cb8-1"></a>mtgenes =<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;^MT-&quot;</span>, <span class="kw">rownames</span>(ctrl[[<span class="st">&quot;RNA&quot;</span>]]),<span class="dt">value =</span> T)</span>
<span id="cb8-2"><a href="downstream.html#cb8-2"></a>ctrl<span class="op">@</span>assays<span class="op">$</span>RNA[mtgenes,] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rowSums</span>()</span></code></pre></div>
<pre><code>##  MT-ND1  MT-ND2  MT-CO1  MT-CO2 MT-ATP8 MT-ATP6  MT-CO3  MT-ND3 MT-ND4L  MT-ND4 
##       0       0       0       0       0       0       0       0       0       0 
##  MT-ND5  MT-ND6  MT-CYB 
##       0       0       0</code></pre>
<p>If reads counts of mitochondrial genes are not equal to zero, we might get a plot look like this.</p>
<p><img src="figs/qc2-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>We can then using dot plot to show the relationship between <code>nCount_RNA</code> and <code>nFeature_RNA</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="downstream.html#cb10-1"></a><span class="co"># FeatureScatter is typically used to visualize feature-feature relationships, but can be used</span></span>
<span id="cb10-2"><a href="downstream.html#cb10-2"></a><span class="co"># for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</span></span>
<span id="cb10-3"><a href="downstream.html#cb10-3"></a></span>
<span id="cb10-4"><a href="downstream.html#cb10-4"></a><span class="kw">FeatureScatter</span>(ctrl, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>)</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Here, we filter cells that have unique feature counts over 2,000 or less than 300</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="downstream.html#cb11-1"></a>ctrl &lt;-<span class="st"> </span><span class="kw">subset</span>(ctrl, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span><span class="dv">300</span> <span class="op">&amp;</span><span class="st"> </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span><span class="dv">2000</span>)</span>
<span id="cb11-2"><a href="downstream.html#cb11-2"></a>stim &lt;-<span class="st"> </span><span class="kw">subset</span>(stim, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span><span class="dv">300</span> <span class="op">&amp;</span><span class="st"> </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span><span class="dv">2000</span>)</span>
<span id="cb11-3"><a href="downstream.html#cb11-3"></a><span class="co"># We can also do the filtering based on percent.mt</span></span>
<span id="cb11-4"><a href="downstream.html#cb11-4"></a><span class="co"># ctrl &lt;- subset(ctrl, subset = percent.mt &lt; 5)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="downstream.html#cb12-1"></a>p1 =<span class="st"> </span><span class="kw">VlnPlot</span>(ctrl, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">pt.size =</span> <span class="fl">0.00001</span>)</span>
<span id="cb12-2"><a href="downstream.html#cb12-2"></a>p2 =<span class="st"> </span><span class="kw">FeatureScatter</span>(ctrl, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">45</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust=</span><span class="dv">1</span>))</span>
<span id="cb12-3"><a href="downstream.html#cb12-3"></a>p1 <span class="op">+</span><span class="st"> </span>p2</span></code></pre></div>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="normalizing-the-data" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Normalizing the data</h3>
<p>After removing unwanted cells from the dataset, the next step is to normalize the data. By default, we employ a global-scaling normalization method “LogNormalize” that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result. Normalized values are stored in <code>object[["RNA"]]@data</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="downstream.html#cb13-1"></a>ifnb.list &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">ctrl =</span> ctrl, <span class="dt">stim =</span> stim)</span>
<span id="cb13-2"><a href="downstream.html#cb13-2"></a></span>
<span id="cb13-3"><a href="downstream.html#cb13-3"></a>ifnb.list &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dt">X =</span> ifnb.list, <span class="dt">FUN =</span> <span class="cf">function</span>(x) {</span>
<span id="cb13-4"><a href="downstream.html#cb13-4"></a>    x &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(x, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="dv">10000</span>)</span>
<span id="cb13-5"><a href="downstream.html#cb13-5"></a>})</span></code></pre></div>
<p>For clarity, in this previous line of code (and in future commands), we provide the default values for certain parameters in the function call. However, this isn’t required and the same behavior can be achieved with:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="downstream.html#cb14-1"></a>x &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(x)</span></code></pre></div>
</div>
<div id="identification-of-highly-variable-features-feature-selection" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Identification of highly variable features (feature selection)</h3>
<p>We next calculate a subset of features that exhibit high cell-to-cell variation in the dataset (i.e, they are highly expressed in some cells, and lowly expressed in others). We and <a href="https://www.nature.com/articles/nmeth.2645">others</a> have found that focusing on these genes in downstream analysis helps to highlight biological signal in single-cell datasets.</p>
<p>Our procedure in Seurat is described in detail <a href="https://www.sciencedirect.com/science/article/pii/S0092867419305598?via%3Dihub">here</a>, and improves on previous versions by directly modeling the mean-variance relationship inherent in single-cell data, and is implemented in the <code>FindVariableFeatures()</code> function. By default, we return 2,000 features per dataset. These will be used in downstream analysis, like PCA.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="downstream.html#cb15-1"></a>ifnb.list &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dt">X =</span> ifnb.list, <span class="dt">FUN =</span> <span class="cf">function</span>(x) {</span>
<span id="cb15-2"><a href="downstream.html#cb15-2"></a>    x &lt;-<span class="st"> </span><span class="kw">FindVariableFeatures</span>(x, <span class="dt">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="dt">nfeatures =</span> <span class="dv">2000</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb15-3"><a href="downstream.html#cb15-3"></a>})</span>
<span id="cb15-4"><a href="downstream.html#cb15-4"></a></span>
<span id="cb15-5"><a href="downstream.html#cb15-5"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb15-6"><a href="downstream.html#cb15-6"></a>top10 &lt;-<span class="st"> </span><span class="kw">head</span>(<span class="kw">VariableFeatures</span>(ifnb.list<span class="op">$</span>ctrl), <span class="dv">10</span>)</span>
<span id="cb15-7"><a href="downstream.html#cb15-7"></a></span>
<span id="cb15-8"><a href="downstream.html#cb15-8"></a><span class="co"># plot variable features with and without labels</span></span>
<span id="cb15-9"><a href="downstream.html#cb15-9"></a>plot1 &lt;-<span class="st"> </span><span class="kw">VariableFeaturePlot</span>(ifnb.list<span class="op">$</span>ctrl)</span>
<span id="cb15-10"><a href="downstream.html#cb15-10"></a><span class="kw">LabelPoints</span>(<span class="dt">plot =</span> plot1, <span class="dt">points =</span> top10, <span class="dt">repel =</span> <span class="ot">TRUE</span>, <span class="dt">max.overlaps =</span> <span class="dv">30</span>)</span></code></pre></div>
<pre><code>## When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<p><img src="Fundamentals-of-scRNASeq-Analysis_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="downstream.html#cb18-1"></a><span class="co"># select features that are repeatedly variable across datasets for integration</span></span>
<span id="cb18-2"><a href="downstream.html#cb18-2"></a>features &lt;-<span class="st"> </span><span class="kw">SelectIntegrationFeatures</span>(<span class="dt">object.list =</span> ifnb.list)</span></code></pre></div>
</div>
</div>
<div id="perform-integration" class="section level2">
<h2><span class="header-section-number">3.4</span> Perform integration</h2>
<p><code>Scaling the data</code> is</p>
<p>We then identify anchors using the <code>FindIntegrationAnchors()</code> function, which takes a list of Seurat objects as input, and use these anchors to integrate the two datasets together with <code>IntegrateData()</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="downstream.html#cb19-1"></a>immune.anchors &lt;-<span class="st"> </span><span class="kw">FindIntegrationAnchors</span>(<span class="dt">object.list =</span> ifnb.list, <span class="dt">anchor.features =</span> features)</span>
<span id="cb19-2"><a href="downstream.html#cb19-2"></a><span class="co"># this command creates an &#39;integrated&#39; data assay</span></span>
<span id="cb19-3"><a href="downstream.html#cb19-3"></a>immune.combined &lt;-<span class="st"> </span><span class="kw">IntegrateData</span>(<span class="dt">anchorset =</span> immune.anchors)</span>
<span id="cb19-4"><a href="downstream.html#cb19-4"></a></span>
<span id="cb19-5"><a href="downstream.html#cb19-5"></a><span class="co"># specify that we will perform downstream analysis on the corrected data note that the</span></span>
<span id="cb19-6"><a href="downstream.html#cb19-6"></a><span class="co"># original unmodified data still resides in the &#39;RNA&#39; assay</span></span>
<span id="cb19-7"><a href="downstream.html#cb19-7"></a><span class="kw">DefaultAssay</span>(immune.combined) &lt;-<span class="st"> &quot;integrated&quot;</span></span></code></pre></div>
<p>The above step will take around 10 minutes. To save time, let’s load the processed data directory.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="downstream.html#cb20-1"></a><span class="kw">load</span>(<span class="st">&quot;./data/immune.combined.Rdata&quot;</span>)</span></code></pre></div>
</div>
<div id="scaling-the-data" class="section level2">
<h2><span class="header-section-number">3.5</span> Scaling the data</h2>
<p>Next, we apply a linear transformation (‘scaling’) that is a standard pre-processing step prior to dimensional reduction techniques like PCA. The ScaleData() function:</p>
<p>Shifts the expression of each gene, so that the mean expression across cells is 0
Scales the expression of each gene, so that the variance across cells is 1
This step gives equal weight in downstream analyses, so that highly-expressed genes do not dominate
The results of this are stored in pbmc[[“RNA”]]<span class="citation">(<span class="citeproc-not-found" data-reference-id="scale.data"><strong>???</strong></span>)</span></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="raw2matrix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 1
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
