<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Pre-processing and QC | Fundamentals of scRNASeq Analysis</title>
  <meta name="description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Pre-processing and QC | Fundamentals of scRNASeq Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  <meta name="github-repo" content="linxy29/Fundamental-scRNA" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Pre-processing and QC | Fundamentals of scRNASeq Analysis" />
  
  <meta name="twitter:description" content="This is the teaching materials for Session 2: Fundamentals of scRNASeq Analysis of 2021 Single Cell Workshop" />
  

<meta name="author" content="Xinyi Lin, Gordon Qian, Joshua Ho contact: jwkho@hku.hk" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="background.html"/>
<link rel="next" href="downstream.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Fundamental scRNASeq</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prerequisites</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-data"><i class="fa fa-check"></i>Download data</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Introduction to Single Cell Technology</a><ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#single-cell-rna-sequencing-scrna-seq"><i class="fa fa-check"></i><b>1.1</b> Single cell RNA sequencing (scRNA-seq)</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#cell-isolation"><i class="fa fa-check"></i><b>1.2</b> Cell isolation</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#barcodes-and-unique-molecular-identifiers-umi"><i class="fa fa-check"></i><b>1.3</b> Barcodes and Unique molecular identifiers (UMI)</a></li>
<li class="chapter" data-level="1.4" data-path="background.html"><a href="background.html#summary-of-widely-used-scrna-seq-technologies"><i class="fa fa-check"></i><b>1.4</b> Summary of widely used scRNA-seq technologies</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="raw2matrix.html"><a href="raw2matrix.html"><i class="fa fa-check"></i><b>2</b> Pre-processing and QC</a><ul>
<li class="chapter" data-level="2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#fastq-file"><i class="fa fa-check"></i><b>2.1</b> FASTQ file</a></li>
<li class="chapter" data-level="2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#cell-ranger"><i class="fa fa-check"></i><b>2.2</b> Cell Ranger</a><ul>
<li class="chapter" data-level="2.2.1" data-path="raw2matrix.html"><a href="raw2matrix.html#general-workflow"><i class="fa fa-check"></i><b>2.2.1</b> General workflow</a></li>
<li class="chapter" data-level="2.2.2" data-path="raw2matrix.html"><a href="raw2matrix.html#output-file"><i class="fa fa-check"></i><b>2.2.2</b> Output file</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="raw2matrix.html"><a href="raw2matrix.html#starsolo"><i class="fa fa-check"></i><b>2.3</b> STARsolo</a></li>
<li class="chapter" data-level="2.4" data-path="raw2matrix.html"><a href="raw2matrix.html#doublets"><i class="fa fa-check"></i><b>2.4</b> Doublets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downstream.html"><a href="downstream.html"><i class="fa fa-check"></i><b>3</b> Analysis Using Seurat</a><ul>
<li class="chapter" data-level="3.1" data-path="downstream.html"><a href="downstream.html#seurat-object"><i class="fa fa-check"></i><b>3.1</b> Seurat object</a></li>
<li class="chapter" data-level="3.2" data-path="downstream.html"><a href="downstream.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>3.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="3.3" data-path="downstream.html"><a href="downstream.html#standard-pre-processing-workflow"><i class="fa fa-check"></i><b>3.3</b> Standard pre-processing workflow</a><ul>
<li class="chapter" data-level="3.3.1" data-path="downstream.html"><a href="downstream.html#qc-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>3.3.1</b> QC and selecting cells for further analysis</a></li>
<li class="chapter" data-level="3.3.2" data-path="downstream.html"><a href="downstream.html#normalizing-the-data"><i class="fa fa-check"></i><b>3.3.2</b> Normalizing the data</a></li>
<li class="chapter" data-level="3.3.3" data-path="downstream.html"><a href="downstream.html#identification-of-highly-variable-features-feature-selection"><i class="fa fa-check"></i><b>3.3.3</b> Identification of highly variable features (feature selection)</a></li>
<li class="chapter" data-level="3.3.4" data-path="downstream.html"><a href="downstream.html#scaling-the-data"><i class="fa fa-check"></i><b>3.3.4</b> Scaling the data</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="downstream.html"><a href="downstream.html#perform-linear-dimensional-reduction"><i class="fa fa-check"></i><b>3.4</b> Perform linear dimensional reduction</a></li>
<li class="chapter" data-level="3.5" data-path="downstream.html"><a href="downstream.html#cluster-the-cells"><i class="fa fa-check"></i><b>3.5</b> Cluster the cells</a></li>
<li class="chapter" data-level="3.6" data-path="downstream.html"><a href="downstream.html#run-non-linear-dimensional-reduction-umaptsne"><i class="fa fa-check"></i><b>3.6</b> Run non-linear dimensional reduction (UMAP/tSNE)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fundamentals of scRNASeq Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="raw2matrix" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Pre-processing and QC</h1>
<div id="fastq-file" class="section level2">
<h2><span class="header-section-number">2.1</span> FASTQ file</h2>
<p>Raw RNA-sequencing data might be in a fastq file. It is a text-based format used for storing read sequences represented by single-letter codes. A sequence in FASTA format begins with a readID started with (“@”) symbol, then lines of sequence data, a separator which is simply a plus (+) sign, and the base call quality scores. It is represented in the following format:</p>
<pre><code>&gt;ReadID
READ SEQUENCE
+
SEQUENCING QUALITY SCORES</code></pre>
<p>For more details about FASTQ files: <a href="https://sapac.support.illumina.com/bulletins/2016/04/fastq-files-explained.html" class="uri">https://sapac.support.illumina.com/bulletins/2016/04/fastq-files-explained.html</a></p>
<p>Generally fastq files are pre-processed using quality control tools, such as <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>. This outputs a series of metrics assessing the quality of sequence reads. Some of these metrics include:</p>
<pre><code>1. Per base sequence quality
2. Per sequence quality scores
3. Per base sequence content
4. Per sequence GC content
5. Per base N content
6. Sequence Duplication Levels
7. Overrepresented Sequences
8. Adapter Content
9. Kmer Content</code></pre>
<p>For more details about FastQC: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
</div>
<div id="cell-ranger" class="section level2">
<h2><span class="header-section-number">2.2</span> Cell Ranger<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></h2>
<p>Figures and contents in this sub-chapter are modified from <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger Support Pages</a>.</p>
<p><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger</a> is a set of analysis pipelines that process Chromium single-cell data to align reads, generate feature-barcode matrices, perform clustering and other secondary analysis, and more. It help us to generate the RNA reads count matrix we will used in chapter <a href="downstream.html#downstream">3</a>.</p>
<p>Some concepts:</p>
<ul>
<li><p><strong>GEM well</strong> (formerly <strong>GEM group</strong>): A set of partitioned cells (Gelbeads-in-Emulsion) from a single 10x Chromium™ Chip channel. One or more sequencing <strong>libraries</strong> can be derived from a <strong>GEM well</strong>.</p></li>
<li><p><strong>Library</strong> (or <strong>Sequencing Library</strong>): A 10x-barcoded sequencing library prepared from a single <strong>GEM well</strong>. With Feature Barcode or V (D) J assays, it is possible to create multiple libraries from the same <strong>GEM well</strong>. The library types may include Gene Expression, Antibody Capture, CRISPR Guide Capture, TCR-enrichment, etc.</p></li>
<li><p><strong>Sequencing Run</strong> (or <strong>Flowcell</strong>): A flowcell containing data from one sequencing instrument run.</p></li>
</ul>
<div id="general-workflow" class="section level3">
<h3><span class="header-section-number">2.2.1</span> General workflow</h3>
<p>Single-cell data getting from Chromium 10X pipeline can be processed through following workflow using cellranger.</p>
<div class="figure" style="text-align: center"><span id="fig:cellranger"></span>
<img src="figs/cellranger_workflow.png" alt="Cellranger Workflow (taken from the cellranger website)" width="100%" />
<p class="caption">
Figure 2.1: Cellranger Workflow (taken from the cellranger website)
</p>
</div>
</div>
<div id="output-file" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Output file<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></h3>
<p>Following is the output folder we will get from cellranger. The <code>outs</code> folder contains the final pipeline output files which including what we want for downstream analysis.</p>
<div class="figure" style="text-align: center"><span id="fig:cellranger2"></span>
<img src="figs/cellranger_output.png" alt="Overview of the folder generate from cellranger" width="100%" />
<p class="caption">
Figure 2.2: Overview of the folder generate from cellranger
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:cellranger3"></span>
<img src="figs/cellranger_output2.png" alt="Overview of the outs folder" width="90%" />
<p class="caption">
Figure 2.3: Overview of the outs folder
</p>
</div>
<p>Above is what we can find in the <code>outs</code> folder. It contains some summary information of the sequencing data, the annotated read sequences, and gene expression matrices we commonly work on. Following is some important output files we want to look at.</p>
<ul>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices">Matrices</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/summary">Web Summary .html</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/analysis">Secondary Analysis CSV</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam">BAM</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/molecule_info">Molecule Info (h5)</a></li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/what-is-loupe-cell-browser">Loupe File (.cloupe)</a></li>
</ul>
<div id="matrices" class="section level5 unnumbered">
<h5>Matrices</h5>
<p>When sequencing, Chromium 10X not only sequences transcriptome, but also sequences any possible molecular. This leads to existence of <code>background barcodes</code>. <strong>Cell associated barcodes</strong> are those barcodes that cellranger believes are marking transcriptome coming from real cells instead of background.</p>
<p>For different version of cellranger, different algorithms are used to detect cell associated barcodes. The general idea is that barcodes for cells should have significantly more transcript counts associated with them than the background barcodes. For more information: <a href="https://kb.10xgenomics.com/hc/en-us/articles/115003480523-How-are-barcodes-classified-as-cell-associated-" class="uri">https://kb.10xgenomics.com/hc/en-us/articles/115003480523-How-are-barcodes-classified-as-cell-associated-</a>.</p>
<p>The cellranger pipeline outputs two types of feature-barcode matrices</p>
<ul>
<li><p><strong>Unfiltered feature-barcode matrix</strong>: is stored under <code>raw_feature_bc_matrix</code> (1-1) folder. Including every barcode from the fixed list of known-good barcode sequences that has at least 1 read, which means it includes background and cell associated barcodes.</p></li>
<li><p><strong>Filtered feature-barcode matrix</strong>: is stored under <code>filtered_feature_bc_matrix</code> (1-2) folder. Including only detected cell associated barcodes.</p></li>
</ul>
<p>Both <code>raw_feature_bc_matrix</code> and <code>filtered_feature_bc_matrix</code> contains three files<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>.</p>
<ul>
<li><p>The <strong>matrix.mtx.gz</strong> file stores reads count as sparse matrices where each row indicates one gene in scRNA-seq data and each column indicates one cell. The information of row indices is stored in the <strong>features.tsv.gz</strong> file while the information of column indicates is stored in the <strong>barcodes.tsv.gz</strong> file.</p></li>
<li><p>The <strong>features.tsv.gz</strong> file corresponds to row indices. In scRNA-seq data, features are actually genes. The features.tsv.gz file contains three columns:</p></li>
</ul>
<pre><code>feature ID: gene_id in the annotation field of the reference GTF
name: gene_name in the annotation field of the reference GTF. If no gene_name field is present in the reference GTF, gene name is equivalent to gene ID.
type of feature: describe features&#39; types, can be one of Gene Expression, Antibody Capture, CRISPR, or CUSTOM. For scRNA-seq data, it will be Gene Expression.</code></pre>
<ul>
<li>The <strong>barcodes.tsv.gz</strong> corresponds to column indices. It contains the barcode of each column. More details on the barcode sequence format are available in the <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam">barcoded BAM section</a>.</li>
</ul>
</div>
<div id="web-summary-.html" class="section level5 unnumbered">
<h5>Web Summary .html</h5>
<p>A summary HTML file contains summary metrics and automated secondary analysis results. If an issue was detected during the pipeline run, an alert appears on this page.</p>
<p>This html files includes two parts, <code>SUMMARY</code> and <code>ANALYSIS</code>. You can also click the ‘?’ in the top of each dashboard for more information on each metric.</p>
<ol style="list-style-type: decimal">
<li>The <strong>SUMMARY</strong> metrics describe sequencing quality and various characteristics of the detected cells. It includes the number of cells detected, the mean reads per cell, and the median genes detected per cell are prominently displayed near the top of the page.</li>
</ol>
<div class="figure" style="text-align: center"><span id="fig:cellranger4"></span>
<img src="figs/cells_summary.png" alt="Cells dashboard (modified from the cellranger website)" width="60%" />
<p class="caption">
Figure 2.4: Cells dashboard (modified from the cellranger website)
</p>
</div>
<p style="padding-left:2em;">
The Barcode Rank Plot under the “Cells” dashboard shows the distribution of barcode counts and which barcodes were inferred to be associated with cells. The y-axis is the number of UMI counts mapped to each barcode and the x-axis is the number of barcodes below that value. A steep drop-off is indicative of good separation between the cell-associated barcodes and the barcodes associated with empty partitions. Barcodes can be determined to be cell-associated based on their UMI count or by their RNA profiles, therefore some regions of the graph can contain both cell-associated and background-associated barcodes. The color of the graph represents the local density of barcodes that are cell-associated.
</p>
<div class="figure" style="text-align: center"><span id="fig:cellranger5"></span>
<img src="figs/seq_map.png" alt="Sequencing and Mapping dashboards (modified from the cellranger website)" width="60%" />
<p class="caption">
Figure 2.5: Sequencing and Mapping dashboards (modified from the cellranger website)
</p>
</div>
<p style="padding-left:2em;">
<p>Other numbers used for evaluation<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>:</p>
<ul>
<li><p><code>Estimated Number of Cells</code>: 500-10,000</p></li>
<li><p><code>Mean Reads per Cell</code>: 20,000 reads/cell minimum recommended reads per cell around 10,000</p></li>
<li><p><code>Valid barcodes</code>: greater than 75%</p></li>
<li><p><code>Q30 bases in RNA read</code>: ideally greater than 65%<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p></li>
<li><p><code>Reads mapped confidently to transcriptome</code>: ideally greater than 30%</p></li>
<li><p><code>Reads mapped antisense to gene</code>: ideally smaller than 10%</p></li>
</ul>
</p>
<ol start="2" style="list-style-type: decimal">
<li>The <strong>ANALYSIS</strong> metrics contains the automated secondary analysis including the following:</li>
</ol>
<pre><code>A dimensional reduction analysis which projects the cells into a 2-D space (t-SNE)
An automated clustering analysis which groups together cells that have similar expression profiles
A list of genes that are differentially expressed between the selected clusters
A plot showing the effect of decreased sequencing depth on observed library complexity
A plot showing the effect of decreased sequencing depth on median genes per cell detected</code></pre>
</div>
<div id="bam" class="section level5 unnumbered">
<h5>BAM</h5>
<p>The <strong>BAM file</strong>, contains information about mapped reads, in a binary compressed format. It is consisted of a optional header section and an alignment section. If there is a a header section, it is distinguished from the alignment section by ‘@’ in the first column and will be prior to the alignment section.</p>
<p>When de-compressed into a SAM file, information is stored as a tab seperated table with some standard columns and some generated by the Cell Ranger software specifically. The Cell Ranger specific columns contains information related to <code>BAM Barcode</code>, <code>BAM Alignment</code> and <code>Feature Barcode</code>.<a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a> The standard columns corresponds to:</p>
<pre><code>QNAME : read name (generally will include UMI barcode if applicable)
FLAG : number tag indicating the “type” of alignment, link to explanation of all possible “types”
RNAME : reference sequence name (i.e. chromosome read is mapped to).
POS : leftmost mapping position
MAPQ : Mapping quality
CIGAR : string indicating the matching/mismatching parts of the read (may include soft-clipping).
RNEXT : reference name of the mate/next read
PNEXT : POS for mate/next read
TLEN : Template length (length of reference region the read is mapped to)
SEQ : read sequence
QUAL : read quality</code></pre>
<p>BAM files can be viewed using SAMtools.</p>
<pre><code>samtools view output.bam</code></pre>
<p>To do trajectory analysis based on RNA velocity, bam file is needed.</p>
</div>
<div id="molecule-info-h5" class="section level5 unnumbered">
<h5>Molecule Info (h5)</h5>
<p>It is an HDF5 file containing per-molecule information for all molecules that contain a valid barcode and valid UMI and were assigned with high confidence to a gene or Feature Barcode. This HDF5 file contains data corresponding to the observed molecules, as well as data about the libraries and feature set(s) used. The structure of this file is:</p>
<div class="figure" style="text-align: center"><span id="fig:cellranger6"></span>
<img src="figs/hdf5.png" alt="HDF5 File Hierarchy(taken from the cellranger website)" width="80%" />
<p class="caption">
Figure 2.6: HDF5 File Hierarchy(taken from the cellranger website)
</p>
</div>
</div>
<div id="secondary-analysis-csv" class="section level5 unnumbered">
<h5>Secondary Analysis CSV</h5>
<p>Several CSV files which contain automated secondary analysis results. It contains information about <code>Dimensionality Reduction</code>, <code>t-SNE</code>, <code>UMAP</code>, <code>Clustering</code> and <code>Differential Expression</code>. It is also visualized in the <code>ANALYSIS</code> metrics through <strong>Web Summary .html</strong> file.</p>
</div>
<div id="loupe-file-.cloupe" class="section level5 unnumbered">
<h5>Loupe File (.cloupe)</h5>
<p>Loupe Browser is a desktop application that provides interactive visualization functionality to analyze data from different 10x Genomics solutions. It can help to <code>Finding Cells of Interest</code>, <code>Finding Significant Genes</code>, <code>Identifying Cell Types</code>, <code>Exploring Substructure</code>, <code>Exploring Cell Subtypes</code>, <code>Integrated Gene Expression and V (D) J Analysis</code>, <code>Sharing Results</code>.</p>
<p>A Loupe file can be viewed through Loupe Browser. It is embedded with the following information:</p>
<ul>
<li><p>Gene expression information for cells in the sample.</p></li>
<li><p>Various gene expression-based clustering information for the cells, including t-SNE and UMAP projections and differential gene expression.</p></li>
<li><p>Gene annotation information from the transcriptome reference.</p></li>
</ul>
</div>
</div>
</div>
<div id="starsolo" class="section level2">
<h2><span class="header-section-number">2.3</span> STARsolo</h2>
<p>The STARsolo <span class="citation">(Kaminow, Yunusov, and Dobin <a href="#ref-kaminow2021starsolo" role="doc-biblioref">2021</a>)</span> is a analysis toool designed for droplet single cell RNA sequencing data (e.g. 10X Genomics Chromium System) built directly into STAR code. The inputs of STARsolo is a FASTQ file, and it can gives nearly identical gene counts in the same format as Cell Ranger but ~10 times faster.</p>
<p>The STARsolo program outputs a large amount of files reflecting details of the read alignment process. We will only discuss some of the key files here.</p>
<div class="figure" style="text-align: center"><span id="fig:starsolo"></span>
<img src="figs/starsolo.png" alt="STARsolo Output" width="100%" />
<p class="caption">
Figure 2.7: STARsolo Output
</p>
</div>
<ol style="list-style-type: decimal">
<li><p>The <strong>BAM file</strong>, contains information about mapped reads which is quiet similar to the BAM file generated through Cell Ranger.</p></li>
<li><p><strong>Alignment summary files</strong>, <code>Features.stats</code> and <code>Summary.csv</code>, contains information about basic mapping details. This can serve as an easy preliminary quality control check of the alignment process.</p></li>
<li><p>The <strong>feature matrix file</strong>, <code>matrix.mtx</code>, contains information about the counts of genes mapped in each individual cell. The column names corresponds to each individual cell barcode, and the row names corresponds to all annotated genes. Due to the large size of this data, it is stored as a sparse matrix.</p></li>
<li><p><strong>Auxiliary files</strong>, <code>barcodes.tsv</code> and <code>features.tsv</code>, provide extra metadata important for downstream analysis. These files along with the matrix file serve the same role as <code>Matrices</code> part in Cell Ranger output and are required for analysis in gold standard scRNA-seq data analysis package, seurat.</p></li>
</ol>
</div>
<div id="doublets" class="section level2">
<h2><span class="header-section-number">2.4</span> Doublets<a href="#fn8" class="footnote-ref" id="fnref8"><sup>8</sup></a></h2>
<p>Doublets are artificial libraries generated from two cells even though it is designed to be generated from only one cell. It usually arise due to errors in cell sorting or capture.</p>
<p>Several experimental strategies can be used to remove doublets:</p>
<ul>
<li><p>based on natural genetic variation between different donor individuals <span class="citation">(Kang et al. <a href="#ref-kang2018multiplexed" role="doc-biblioref">2018</a>)</span>.</p></li>
<li><p>mark a subset of cells (e.g., all cells from one sample) with an antibody conjugated to a different oligonucleotide. Upon pooling, libraries that are observed to have different oligonucleotides are considered to be doublets and removed <span class="citation">(Bach et al. <a href="#ref-bach2017differentiation" role="doc-biblioref">2017</a>)</span>.</p></li>
<li><p>infer doublets from the expression profiles alone based on computational approaches like simulation.</p></li>
</ul>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bach2017differentiation">
<p>Bach, Karsten, Sara Pensa, Marta Grzelak, James Hadfield, David J Adams, John C Marioni, and Walid T Khaled. 2017. “Differentiation Dynamics of Mammary Epithelial Cells Revealed by Single-Cell Rna Sequencing.” <em>Nature Communications</em> 8 (1): 1–11.</p>
</div>
<div id="ref-kaminow2021starsolo">
<p>Kaminow, Benjamin, Dinar Yunusov, and Alexander Dobin. 2021. “STARsolo: Accurate, Fast and Versatile Mapping/Quantification of Single-Cell and Single-Nucleus Rna-Seq Data.” <em>bioRxiv</em>.</p>
</div>
<div id="ref-kang2018multiplexed">
<p>Kang, Hyun Min, Meena Subramaniam, Sasha Targ, Michelle Nguyen, Lenka Maliskova, Elizabeth McCarthy, Eunice Wan, et al. 2018. “Multiplexed Droplet Single-Cell Rna-Sequencing Using Natural Genetic Variation.” <em>Nature Biotechnology</em> 36 (1): 89–94.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="2">
<li id="fn2"><p>For more information: <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger</a><a href="raw2matrix.html#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>For more information: <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/overview" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/overview</a><a href="raw2matrix.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>The <strong>desktop. ini</strong> file is a hidden file used to store information about the arrangement of a Windows folder.<a href="raw2matrix.html#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>The information here is provided by <a href="https://assets.ctfassets.net/an68im79xiti/163qWiQBTVi2YLbskJphQX/e90bb82151b1cdab6d7e9b6c845e6130/CG000329_TechnicalNote_InterpretingCellRangerWebSummaryFiles_RevA.pdf" class="uri">https://assets.ctfassets.net/an68im79xiti/163qWiQBTVi2YLbskJphQX/e90bb82151b1cdab6d7e9b6c845e6130/CG000329_TechnicalNote_InterpretingCellRangerWebSummaryFiles_RevA.pdf</a><a href="raw2matrix.html#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>Q scores are defined as a property that is logarithmically related to the base calling error probabilities. For more information: <a href="https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf" class="uri">https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf</a><a href="raw2matrix.html#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p>For more details: <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/bam</a><a href="raw2matrix.html#fnref7" class="footnote-back">↩︎</a></p></li>
<li id="fn8"><p>For more information: <a href="https://bioconductor.org/books/release/OSCA/doublet-detection.html" class="uri">https://bioconductor.org/books/release/OSCA/doublet-detection.html</a><a href="raw2matrix.html#fnref8" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="background.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="downstream.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
